packages/babel-plugin-bugfix-safari-class-field-initializer-scope/src/index.ts

```diff
@@ -0,0 +1,92 @@
+import { declare } from "@babel/helper-plugin-utils";
+import { types as t } from "@babel/core";
+import type { NodePath } from "@babel/core";
+
+function needsWrapping(node: t.Node): boolean {
+  if (t.isLiteral(node) && !t.isTemplateLiteral(node)) {
+    return false;
+  }
+
+  if (t.isCallExpression(node) || t.isNewExpression(node)) {
+    return needsWrapping(node.callee) || node.arguments.some(needsWrapping);
+  }
+
+  if (t.isTemplateLiteral(node)) {
+    return node.expressions.some(needsWrapping);
+  }
+
+  if (t.isTaggedTemplateExpression(node)) {
+    return needsWrapping(node.tag) || needsWrapping(node.quasi);
+  }
+
+  if (t.isArrayExpression(node)) {
+    return node.elements.some(needsWrapping);
+  }
+
+  if (t.isObjectExpression(node)) {
+    return node.properties.some(prop => {
+      if (t.isObjectProperty(prop)) {
+        return (
+          needsWrapping(prop.value) ||
+          (prop.computed && needsWrapping(prop.key))
+        );
+      }
+      if (t.isObjectMethod(prop)) {
+        return false;
+      }
+      return false;
+    });
+  }
+
+  if (t.isMemberExpression(node)) {
+    return (
+      needsWrapping(node.object) ||
+      (node.computed && needsWrapping(node.property))
+    );
+  }
+
+  if (t.isFunctionExpression(node) || t.isArrowFunctionExpression(node)) {
```

Comment:
Should we also cover ClassExpression here?

Reply:
Yup, added :+1:

I have confirmed that the `superClass` expression does not seem to be affected by the Safari bug, so we don't need to recurse into that.